File: frontend/types/scheme.ts
export interface SchemeSummary {
  id: string;
  name: string;
  category: string;
  level: string;
  benefit: string;
  source: string;
  status: string;
}

export interface SchemeSearchResult {
  total: number;
  results: SchemeSummary[];
}

export interface Scheme {
  id: string;
  name: string;
  summary: string;
  benefit: string;
  category: string;
  categories: string[];
  level: string;
  requiredDocuments: string[];
  requirementTags: string[];
  tags: string[];
  officialUrl: string | null;
  source: string;
  sourceType: string;
  status: string;
}

export interface SchemeMatch {
  id: string;
  name: string;
  category: string;
  level: string;
  benefit: string;
  satisfied: number;
  total: number;
  matchedTags: string[];
  missingTags: string[];
  officialUrl: string | null;
  source: string;
  status: string;
}

export interface SchemeMatchResult {
  results: SchemeMatch[];
}

File: backend/app/schemas/scheme.py
from pydantic import BaseModel


class SchemeSummary(BaseModel):
    id: str
    name: str
    category: str
    level: str
    benefit: str = ""
    source: str = "myScheme"
    status: str = "needs_verification"


class SchemeSearchResult(BaseModel):
    total: int
    results: list[SchemeSummary] = []


class SchemeRead(BaseModel):
    id: str
    name: str
    summary: str = ""
    benefit: str = ""
    category: str
    categories: list[str] = []
    level: str
    requiredDocuments: list[str] = []
    requirementTags: list[str] = []
    tags: list[str] = []
    officialUrl: str | None = None
    source: str = "myScheme"
    sourceType: str = "dataset"
    status: str = "needs_verification"


class SchemeMatch(BaseModel):
    id: str
    name: str
    category: str
    level: str
    benefit: str = ""
    satisfied: int
    total: int
    matchedTags: list[str] = []
    missingTags: list[str] = []
    officialUrl: str | None = None
    source: str = "myScheme"
    status: str = "needs_verification"


class SchemeMatchResult(BaseModel):
    results: list[SchemeMatch] = []
